### PR TITLE
CLOUDP-121809: ask to regenerate device code on login flow

### PR DESCRIPTION
**Pull request by tibulca**
## Proposed changes

_Jira ticket:_ CLOUDP-121809

Closes #[issue number]

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [x] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/auth/login.go [hunk 1]
```diff
@@ -52,18 +52,44 @@ const (
...
+	message         string
+	defaultResponse bool
+}
+
+func (c *confirmPrompt) confirm() (response bool, err error) {
+	p := &survey.Confirm{
+		Message: c.message,
+		Default: c.defaultResponse,
+	}
+	err = survey.AskOne(p, &response)
```
Comment (blva) [inline internal/cli/auth/login.go:91]: Please add a TODO for us to remember to update this for the telemetry version as per this PR https://github.com/mongodb/mongodb-atlas-cli/pull/1214/files

Comment (blva) [reply internal/cli/auth/login.go:91]: can be a ticket as well IMO

File: internal/cli/auth/login.go [hunk 2]
```diff
@@ -174,30 +200,46 @@ Your code will expire after %.0f minutes.
...
+		opts.AccessToken = accessToken.AccessToken
+		opts.RefreshToken = accessToken.RefreshToken
+		return nil
 	}
-	opts.AccessToken = accessToken.AccessToken
-	opts.RefreshToken = accessToken.RefreshToken
-	return nil
+}
+
+func (opts *LoginOpts) shouldRetryRegister(err error) (retry bool, errSurvey error) {
```
[the remark this answers is not in the document]
Comment (tibulca) [reply internal/cli/auth/login.go:237]: done, thanks for spotting this

File: internal/cli/auth/login.go (hunk 1 above)
Comment (tibulca) [reply internal/cli/auth/login.go:93]: done, added a todo referencing CLOUDP-118532

**Review by blva — APPROVED** [review]

**Pull request merged**